cpu/esp32/periph/Kconfig.i2c

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+if TEST_KCONFIG
+
+choice
+    bool "I2C implementation"
+    depends on MODULE_PERIPH_I2C
+    # necessary to fix driver initialization on esp32
+    default MODULE_ESP_I2C_HW if MODULE_AIP31068
```

Comment:
This dependency isn't clear, where does it come frome? AIP31068 is a liquid crystal I2C/SPI LCD display.

Reply:
The default should always be the software implementation.

Reply:
It comes from here:
https://github.com/RIOT-OS/RIOT/blob/44ebc38caca3ce24914a6b543fb34703a304e660/drivers/aip31068/Makefile.dep#L5-L8

I also don't like this, but we are trying to match the Makefile version. @HendrikVE  why is this dependency needed?

Reply:
Hm, @HendrikVE does this mean that AIP31068 is only working with the hardware implementation of I2C?

Reply:
BTW, out of topic, I have open PR #12665 since two years which simply extends the existing HD44170 driver by the I2C IO expander PCF857x. I didn't take a look into PR #14966 in detail.

Reply:
I'm not entirely sure what the problem was back then. I don't have the hardware with me today so I'll try it again at home this evening and report back

Reply:
@gschorcht I did a quick test and I can confirm that this problem still exists. When using `esp_i2c_hw` then the initialization always work. But without it it will fail in 80% of the cases with `rc = -6`. 

It happens here:
https://github.com/RIOT-OS/RIOT/blob/c84a40abc4511f2bc5b55f0db8e708432e32d161/drivers/aip31068/aip31068.c#L145
It is caused by:
https://github.com/RIOT-OS/RIOT/blob/207964265a459579ef38781be8945366d3cc88a4/cpu/esp_common/periph/i2c_sw.c#L365

Reply:
This is really strange, I use the I2C software implementation on ESP32 and EP8266 very extensively with a large number of different devices and never had any problems. My impression was rather that the hardware implementation is a bit unstable, the I2C hardware sometimes gets stuck, so there is an extra bus `_i2c_reset_hw` cycle in each `i2c_acquire` and `i2c_release` which generates 20 clock cycles while the SDA line is held LOW, which creates a noticeable delay in each transfer.

Without having the hardware I can't initialize the problem with logic analyzer but it would be very interesting why the I2C software implementation returns with that error in 80 % of the cases.

Reply:
Ok, I have tested the LCD module with the I2C software implementation. On ESP32 I can reproduce that very often it is not possible to communicate with the AIP31068. But the same software implementation is used for ESP8266 where it works reliably. The only difference between ESP32 and ESP8266 is that the GPIOs are used in a different mode for timing reasons. Maybe that makes the difference.

Anyway, using the same mechanism to clear the bus during I2C initialization (sending 10 clock pulses while SDA is low) solves this problem for the software implementation on ESP32. Obviously, the AIP31068 is a special sensitive. I use ESP32 very intensively with a large number of different I2C devices and have never had such problems.

Finally, I would prefer to remove this special case. I will provide a fix for the software implementation for ESP32.

BTW, my module is from Waveshare which has an additional RGB backlight control using a PCA9633DP2 which has to controlled too, otherwise you don't have backlight as in my case. Maybe we should extend your example by the option to enable module `pca9633` to control the backlight.

Reply:
With #17346 in, I pushed e6126904389b0a58c24c0a14fdd1e74eb479064b which removes this strange dependency. Thanks @gschorcht !